src/platform/Linux/bluez/ChipDeviceScanner.cpp

```diff
@@ -131,13 +131,13 @@ CHIP_ERROR ChipDeviceScanner::StartScan(unsigned timeoutMs)
         return CHIP_ERROR_INTERNAL;
     }
 
-    CHIP_ERROR err = chip::DeviceLayer::SystemLayer.StartTimer(timeoutMs, TimerExpiredCallback, static_cast<void *>(this));
+    System::Timer * lTimer = chip::DeviceLayer::SystemLayer.StartTimer(timeoutMs, TimerExpiredCallback, static_cast<void *>(this));
 
-    if (err != CHIP_NO_ERROR)
+    if (lTimer == nullptr)
     {
         ChipLogError(Ble, "Failed to schedule scan timeout.");
         StopScan();
-        return err;
+        return CHIP_ERROR_INTERNAL;
```

Comment:
same here: i believe we should standardize on an error code on start timer failures, maybe an exlicit 'Cannot schedule timer' or a generic 'no resources available'. INTERNAL feels to generic.

Reply:
Acked